control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -118,19 +122,47 @@ func findTagReference(tags []imagev1.TagReference, name string) *imagev1.TagRefe
 	return nil
 }
 
-var CatalogToImage = map[string]string{
-	"certified-operators": "registry.redhat.io/redhat/certified-operator-index:v4.16",
-	"community-operators": "registry.redhat.io/redhat/community-operator-index:v4.16",
-	"redhat-marketplace":  "registry.redhat.io/redhat/redhat-marketplace-index:v4.16",
-	"redhat-operators":    "registry.redhat.io/redhat/redhat-operator-index:v4.16",
+func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullSecret []byte, digestLister registryclient.DigestListerFN) (map[string]string, error) {
+	imageRef := hcp.Spec.ReleaseImage
+	imageConfig, _, _, err := registryclient.GetMetadata(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image metadata: %w", err)
+	}
+
+	version, err := semver.Parse(imageConfig.Config.Labels["io.openshift.release"])
+	if err != nil {
+		return nil, fmt.Errorf("invalid OpenShift release version format: %s", imageConfig.Config.Labels["io.openshift.release"])
+	}
+
+	operators := map[string]string{
+		"certified-operators": fmt.Sprintf("registry.redhat.io/redhat/certified-operator-index:v%s", version.String()),
+		"community-operators": fmt.Sprintf("registry.redhat.io/redhat/community-operator-index:v%s", version.String()),
+		"redhat-marketplace":  fmt.Sprintf("registry.redhat.io/redhat/redhat-marketplace-index:v%s", version.String()),
+		"redhat-operators":    fmt.Sprintf("registry.redhat.io/redhat/redhat-operator-index:v%s", version.String()),
+	}
+
+	for i := 0; i < 4; i++ {
+		_, err = digestLister(ctx, operators["certified-operators"], pullSecret)
+		if err != nil {
+			if !strings.Contains(err.Error(), "manifest unknown") {
+				return operators, err
```

Comment:
why are we returning a valid value with the error
```suggestion
				return nil, err
```

Reply:
yeah we dont need to now, left over from the previous approach.